### PR TITLE
MAT-4592: Adding Stratifications to bundle

### PR DESCRIPTION
**Pull request by gregory-akins**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-4592](https://jira.cms.gov/browse/MAT-4592)
(Optional) Related Tickets:

### Summary
The only relevant changes were to MeasureTranslatorService and MeasureTranslatorServiceTest

The pom changes and the other files were for formatting and some build plugins (Checkstyle and formatting)

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java [hunk 1]
```diff
@@ -13,13 +13,15 @@
...
 import org.hl7.fhir.instance.model.api.IBaseDatatype;
+import org.hl7.fhir.r4.model.BackboneElement;
 import org.hl7.fhir.r4.model.CanonicalType;
 import org.hl7.fhir.r4.model.CodeableConcept;
 import org.hl7.fhir.r4.model.Coding;
 import org.hl7.fhir.r4.model.ContactDetail;
 import org.hl7.fhir.r4.model.ContactPoint;
 import org.hl7.fhir.r4.model.Expression;
 import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.Measure.MeasureGroupStratifierComponent;
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java:20]: This class has quite a few unused imports

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java [hunk 2]
```diff
@@ -57,94 +59,149 @@ public org.hl7.fhir.r4.model.Measure createFhirMeasureForMadieMeasure(Measure ma
...
-        // TODO: Add an extension for measure observations
-      }).collect(Collectors.toList());
-
-    return (MeasureGroupComponent)(new MeasureGroupComponent().setPopulation(measurePopulations)
-        .setId(madieGroup.getId()));
+  private List<MeasureGroupStratifierComponent> buildStratifications(Group madieGroup) {
+    AtomicReference<Extension> extension = new AtomicReference<Extension>();
+    List<MeasureGroupStratifierComponent> measureStratifications = null;
+    if (madieGroup.getStratifications() != null) {
+      AtomicReference<Integer> i = new AtomicReference<>();
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java:120]: Why do we need AtomicReferences?

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java [hunk 3]
```diff
@@ -57,94 +59,149 @@ public org.hl7.fhir.r4.model.Measure createFhirMeasureForMadieMeasure(Measure ma
...
+      i.set(Integer.valueOf(0));
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java:121]: If Atomic is required, How about we use AtomicInteger which has a constructor that initializes with 0 and accepts an int.

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java [hunk 4]
```diff
@@ -57,94 +59,149 @@ public org.hl7.fhir.r4.model.Measure createFhirMeasureForMadieMeasure(Measure ma
...
-            //TODO investigate whether these URLS can change; 
-            //  if they are codified in HAPI FHIR Libraries.. 
-            //  we should define a separate property file with this list 
-            extension.set(
-                new Extension("http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-criteriaReference", 
-                    theValue)) ;
-          }
-        });
-        
+    AtomicReference<String> id = new AtomicReference<String>();
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java:156]: It looks like id is never used

File: src/test/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorServiceTest.java [hunk 5]
```diff
@@ -166,46 +188,99 @@ public void testBuildFhirPopulationGroupsWithAssocations() {
...
+    pops.add(ip1);
+    pops.add(ip2);
+    group.setPopulations(pops);
+    List<Stratification> stratifications = new ArrayList<>();
+    Stratification strat1 = new Stratification();
+    strat1.setDescription(null);
+    stratifications.add(strat1);
+    group.setStratifications(stratifications);
+    List<Group> groups = new ArrayList<>();
+    groups.add(group);
```
Comment (RohitKandimalla) [inline src/test/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorServiceTest.java:262]: Not a big deal, but using Object Builder could make this more readable.

File: src/test/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorServiceTest.java [hunk 6]
```diff
@@ -10,11 +10,14 @@
...
 import org.apache.commons.lang3.time.DateFormatUtils;
 import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Expression;
 import org.hl7.fhir.r4.model.Extension;
 import org.hl7.fhir.r4.model.Measure.MeasureGroupComponent;
 import org.hl7.fhir.r4.model.Measure.MeasureGroupPopulationComponent;
+import org.hl7.fhir.r4.model.Measure.MeasureGroupStratifierComponent;
 import org.hl7.fhir.r4.model.Meta;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
```
Comment (RohitKandimalla) [inline src/test/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorServiceTest.java:21]: Unused statements

**Review by RohitKandimalla — CHANGES_REQUESTED** [review]

File: src/test/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorServiceTest.java (hunk 5 above)
Comment (gregory-akins) [reply src/test/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorServiceTest.java:262]: +1

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java (hunk 2 above)
Comment (gregory-akins) [reply src/main/java/gov/cms/madie/madiefhirservice/services/MeasureTranslatorService.java:120]: final objects in the iteration

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**